sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public int Status { get { throw null; } }
+    }
+}
+namespace System.Net.ClientModel.Core
+{
+    public partial class ClientMessage : System.IDisposable
+    {
+        protected internal ClientMessage(System.Net.ClientModel.Core.MessageRequest request) { }
+        public virtual System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public bool HasResponse { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageClassifier MessageClassifier { get { throw null; } set { } }
+        public virtual System.Net.ClientModel.Core.MessageRequest Request { get { throw null; } }
+        public virtual System.Net.ClientModel.Core.MessageResponse Response { get { throw null; } protected internal set { } }
+        public virtual void Dispose() { }
+        protected virtual void Dispose(bool disposing) { }
+        public void SetProperty(System.Type type, object value) { }
+        public bool TryGetProperty(System.Type type, out object? value) { throw null; }
+    }
+    [System.FlagsAttribute]
+    public enum ErrorBehavior
+    {
+        Default = 0,
+        NoThrow = 1,
+    }
+    public partial interface IJsonModel<out T> : System.Net.ClientModel.Core.IModel<T>
+    {
+        T Read(ref System.Text.Json.Utf8JsonReader reader, System.Net.ClientModel.ModelReaderWriterOptions options);
+        void Write(System.Text.Json.Utf8JsonWriter writer, System.Net.ClientModel.ModelReaderWriterOptions options);
+    }
+    public partial interface IModel<out T>
+    {
+        System.Net.ClientModel.ModelReaderWriterFormat GetWireFormat(System.Net.ClientModel.ModelReaderWriterOptions options);
+        T Read(System.BinaryData data, System.Net.ClientModel.ModelReaderWriterOptions options);
+        System.BinaryData Write(System.Net.ClientModel.ModelReaderWriterOptions options);
+    }
+    public partial class KeyCredentialAuthenticationPolicy : System.Net.ClientModel.Core.PipelinePolicy
+    {
+        public KeyCredentialAuthenticationPolicy(System.Net.ClientModel.KeyCredential credential, string headerName, string? keyPrefix = null) { }
+        public override void Process(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { }
+        public override System.Threading.Tasks.ValueTask ProcessAsync(System.Net.ClientModel.Core.ClientMessage message, System.Net.ClientModel.Core.PipelineEnumerator pipeline) { throw null; }
+    }
+    public partial class MessageClassifier
+    {
+        protected internal MessageClassifier() { }
+        public virtual bool IsError(System.Net.ClientModel.Core.ClientMessage message) { throw null; }
+    }
+    public abstract partial class MessageHeaders
+    {
+        protected MessageHeaders() { }
+        public abstract int Count { get; }
+        public abstract void Add(string name, string value);
+        public abstract bool Remove(string name);
```

Comment:
Should this be `TryRemove` to allow for scenarios where the header is not present rather than this causing an error in that scenario or not being sure whether it existed or not prior to calling Remove?

Reply:
Removing an item that does not exist is not an error. It just returns false. Try List<T>.Remove.

Reply:
Do we think there could be scenarios where we'd want to know that a header existed or not prior to removal?

Reply:
> Do we think there could be scenarios where we'd want to know that a header existed or not prior to removal?

Yes. You can check the return value if you want to know.

Reply:
ah - missed the return value.

Reply:
I tried to unify Azure.Core's RequestHeaders and ResponseHeaders into a single type with this MessageHeaders type, overall to reduce the number of types in ClientModel.  I still have some open questions about whether this is the right approach, some related to perf, so expect this to change before we're done.  Any feedback shared now will be incorporated into the revisions of this type, so thank you in advance for them!

Reply:
One concern I had about `MessageHeaders` is that `MessageResponseHeaders` doesn't implement a lot of it.